var/spack/repos/builtin/packages/hiop/package.py

```diff
@@ -232,42 +230,54 @@ def cmake_args(self):
 
         return args
 
+    def _run_test_example(self, test_exe, options):
+        """Run the provided test executable with the provided options."""
+
+        exe_path = os.path.join(self.prefix.bin, test_exe)
+        if not os.path.exists(exe_path):
+            raise SkipTest("{0} is not installed".format(test_exe))
+
+        exe = which(exe_path)
+        exe(*options)
+
     # If testing on a cluster without access to home directory in a job, you may
     # set the following environment variables to prevent related errors:
     #
     # export SPACK_USER_CACHE_PATH=/tmp/spack
     # export SPACK_DISABLE_LOCAL_CONFIG=true
-    def test(self):
-        if not self.spec.satisfies("@develop") or not os.path.isdir(self.prefix.bin):
-            tty.info("Skipping: checks not installed in bin for v{0}".format(self.version))
-            return
+    def test_nlpmdsex1(self):
+        """run NlpMDsEx1.exe selfchecks"""
+        # TODO: These tests should build and run against the installed software.
+        # TODO: Set `test_requires_compiler = True` to pick up them and cmake.
+        # TODO: These tests should *never* run in the installation prefix.
 
         tests = [
-            ["NlpMdsEx1.exe", "400", "100", "0", "-selfcheck"],
-            ["NlpMdsEx1.exe", "400", "100", "1", "-selfcheck"],
-            ["NlpMdsEx1.exe", "400", "100", "0", "-empty_sp_row", "-selfcheck"],
+            ("NlpMdsEx1.exe", ["400", "100", "0", "-selfcheck"]),
+            ("NlpMdsEx1.exe", ["400", "100", "1", "-selfcheck"]),
+            ("NlpMdsEx1.exe", ["400", "100", "0", "-empty_sp_row", "-selfcheck"]),
         ]
+        for exe, args in tests:
+            with test_part(
```

Comment:
I am unfamiliar with `test_part` here.

Reply:
It's the new API for tests.  The docs describe it in the section at  https://spack.readthedocs.io/en/latest/packaging_guide.html#adding-stand-alone-tests.

Reply:
Ok I will probably have to re-visit this when adding tests to ExaGO, but I really appreciate this :)